src/Agent.Listener/JobDispatcher.cs

```diff
@@ -632,17 +641,33 @@ private async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockTok
                     }
                     else
                     {
-                        // retry till reach lockeduntil
-                        remainingTime = request.LockedUntil.Value - DateTime.UtcNow;
+                        // retry till reach lockeduntil + 5 mins oauth token time skew limit + 5 mins extra buffer for the delay of request monitor job on server side.
```

Comment:
nit: 5 minutes is probably sufficient. 5 + 5 assumes two different exceptional cases at once.